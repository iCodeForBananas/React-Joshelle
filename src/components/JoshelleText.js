// @flow
import React from "react";
import "./JoshelleText.css";

type Props = {};

type State = {
  seconds: number
};

class JoshelleText extends React.Component<Props, State> {
  state = {
    seconds: 0
  };

  componentDidMount() {
    this.incrementSeconds();
  }

  incrementSeconds() {
    this.setState(
      {
        seconds: this.state.seconds + 1
      },
      () => {
        setTimeout(() => {
          this.incrementSeconds();
        }, 1000);
      }
    );
  }

  render() {
    var chars = `You have been Joshelleing for ${this.state.seconds} seconds!`.split("");

    return (
      <span className="wavetext">
        {chars.map((char, index) => (
          <span key={char + index}>{char}</span>
        ))}
      </span>
    );
  }
}

export default JoshelleText;
