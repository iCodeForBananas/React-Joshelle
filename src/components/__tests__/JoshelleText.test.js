import React from "react";
import JoshelleText from "../JoshelleText";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<JoshelleText />).toJSON();
  expect(tree).toMatchSnapshot();
});
