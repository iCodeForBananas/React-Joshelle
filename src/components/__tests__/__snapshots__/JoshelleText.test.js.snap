// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="wavetext"
>
  <span>
    Y
  </span>
  <span>
    o
  </span>
  <span>
    u
  </span>
  <span>
     
  </span>
  <span>
    h
  </span>
  <span>
    a
  </span>
  <span>
    v
  </span>
  <span>
    e
  </span>
  <span>
     
  </span>
  <span>
    b
  </span>
  <span>
    e
  </span>
  <span>
    e
  </span>
  <span>
    n
  </span>
  <span>
     
  </span>
  <span>
    J
  </span>
  <span>
    o
  </span>
  <span>
    s
  </span>
  <span>
    h
  </span>
  <span>
    e
  </span>
  <span>
    l
  </span>
  <span>
    l
  </span>
  <span>
    e
  </span>
  <span>
    i
  </span>
  <span>
    n
  </span>
  <span>
    g
  </span>
  <span>
     
  </span>
  <span>
    f
  </span>
  <span>
    o
  </span>
  <span>
    r
  </span>
  <span>
     
  </span>
  <span>
    1
  </span>
  <span>
     
  </span>
  <span>
    s
  </span>
  <span>
    e
  </span>
  <span>
    c
  </span>
  <span>
    o
  </span>
  <span>
    n
  </span>
  <span>
    d
  </span>
  <span>
    s
  </span>
  <span>
    !
  </span>
</span>
`;
